devlib/target.py

```diff
@@ -361,7 +361,8 @@ def ps(self, **kwargs):
 
     def file_exists(self, filepath):
         command = 'if [ -e \'{}\' ]; then echo 1; else echo 0; fi'
-        return boolean(self.execute(command.format(filepath)).strip())
+        output = self.execute(command.format(filepath), as_root=True)
```

Comment:
This change makes `file_exists` unusable on unrooted devices. Should be

    output = self.execute(command.format(filepath), as_root=self.is_rooted)